src/hotspot/cpu/x86/x86_64.ad

```diff
@@ -8595,10 +9724,24 @@ instruct rolI_rReg_Var(rRegI dst, rcx_RegI shift, rFlagsReg cr)
   ins_pipe(ialu_reg_reg);
 %}
 
+// Rotate Left by variable
+instruct rolI_rReg_Var_ndd(rRegI dst, rRegI src, rcx_RegI shift, rFlagsReg cr)
+%{
+  predicate(UseAPX && n->bottom_type()->basic_type() == T_INT);
+  match(Set dst (RotateLeft src shift));
+  effect(KILL cr);
+
+  format %{ "eroll    $dst, $src, $shift\t# rotate left (int ndd)" %}
+  ins_encode %{
+    __ eroll($dst$$Register, $src$$Register, false);
+  %}
+  ins_pipe(ialu_reg_reg);
+%}
+
 // Rotate Right by constant.
 instruct rorI_immI8_legacy(rRegI dst, immI8 shift, rFlagsReg cr)
 %{
-  predicate(!VM_Version::supports_bmi2() && n->bottom_type()->basic_type() == T_INT);
+  predicate(!UseAPX && !VM_Version::supports_bmi2() && n->bottom_type()->basic_type() == T_INT);
```

Comment:
Don't need to change anything for non bmi2 platforms. The original predicate can be kept as is. This applies to all rorI, rolI, rorL, rolL.

Reply:
Please see the updated code removing rorI, rolI, rorL, rolL for non bmi2 platforms.